assets/js/themes.js

```diff
@@ -33,3 +36,17 @@ function set_mode(bool) {
         toggle_theme.children[0].setAttribute('class', 'icon ion-ios-moon');
     }
 }
+
+function update_mode (mode) {
+    if (mode == 'true' /* for backwards compatibility */ || mode == 'dark') {
+        // If dark mode preference set
+        set_mode(true);
+    }
+    else if (mode != 'light' && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Comment:
Should this be `mode == undefined`? Since when first visiting a page without `prefers-color-scheme`, this will load the dark theme since `undefined != 'true'` is true.